solr-search/src/main/java/edu/unc/lib/dl/search/solr/util/FacetFieldUtil.java

```diff
@@ -41,69 +46,128 @@
+                .map(toFqFunct)
+                .collect(Collectors.joining(" OR "));
+        solrQuery.addFilterQuery(fq);
+    }
+
+    private Function<SearchFacet, String> cutoffFacetToFq = (inFacet) -> {
+        CutoffFacet facet = (CutoffFacet) inFacet;
         List<HierarchicalFacetNode> facetNodes = facet.getFacetNodes();
         CutoffFacetNode endNode = (CutoffFacetNode) facetNodes.get(facetNodes.size() - 1);
         String solrFieldName = solrSettings.getFieldName(facet.getFieldName());
 
-        StringBuilder filterQuery = new StringBuilder();
+        StringBuilder filterQuery = new StringBuilder("(");
         filterQuery.append(solrFieldName).append(":").append(endNode.getTier()).append(",");
         if (!endNode.getSearchKey().equals("*")) {
             filterQuery.append(SolrSettings.sanitize(endNode.getSearchKey()));
         } else {
             filterQuery.append('*');
         }
-        solrQuery.addFilterQuery(filterQuery.toString());
 
         if (facet.getCutoff() != null) {
-            filterQuery = new StringBuilder();
-            filterQuery.append('!').append(solrFieldName).append(':').append(facet.getCutoff()).append(',').append('*');
-            solrQuery.addFilterQuery(filterQuery.toString());
+            filterQuery.append(" AND !").append(solrFieldName).append(':').append(facet.getCutoff())
+                .append(',').append('*');
         }
+        return filterQuery.append(')').toString();
+    };
 
-        if (facet.getFacetCutoff() != null) {
-            solrQuery.setFacetPrefix(solrFieldName, facet.getFacetCutoff() + ",");
-        }
-    }
-
-    private void addMultivaluedFacetValue(MultivaluedHierarchicalFacet facet, SolrQuery solrQuery) {
+    private Function<SearchFacet, String> multivaluedFacetToFq = (inFacet) -> {
+        MultivaluedHierarchicalFacet facet = (MultivaluedHierarchicalFacet) inFacet;
         StringBuilder filterQuery = new StringBuilder();
         String solrFieldName = solrSettings.getFieldName(facet.getFieldName());
 
         filterQuery.append(solrFieldName).append(":").append(
                 SolrSettings.sanitize(facet.getSearchValue())).append(",*");
-        solrQuery.addFilterQuery(filterQuery.toString());
-
-        solrQuery.add("f." + solrFieldName + ".facet.prefix", facet.getPivotValue());
-    }
-
-    private void addGenericFacetValue(GenericFacet facet, SolrQuery solrQuery) {
-        solrQuery.addFilterQuery(solrSettings.getFieldName(facet.getFieldName()) + ":\""
-                + SolrSettings.sanitize(facet.getSearchValue()) + "\"");
-    }
-
-    private void addCaseInsensitiveFacetValue(CaseInsensitiveFacet facet, SolrQuery solrQuery) {
-        solrQuery.addFilterQuery(solrSettings.getFieldName(facet.getSearchName()) + ":\""
-                + SolrSettings.sanitize(facet.getSearchValue()) + "\"");
-    }
+        return filterQuery.toString();
+    };
+
+    private Function<SearchFacet, String> genericFacetToFq = (facet) -> {
```

Comment:
What's a genericFacet?

Reply:
`edu.unc.lib.dl.search.solr.model.GenericFacet`
Any facet field not mapped to a specific class is a generic facet. For example, PARENT_COLLECTION is a generic facet, while CONTENT_TYPE is a MultivaluedHierarchicalFacet. You can see the classes used per facet in search.properties https://github.com/UNC-Libraries/Carolina-Digital-Repository/blob/main/access/src/main/resources/search.properties#L100